src/node.cc

```diff
@@ -4363,7 +4403,9 @@ void EmitBeforeExit(Environment* env) {
     FIXED_ONE_BYTE_STRING(env->isolate(), "beforeExit"),
     process_object->Get(exit_code)->ToInteger(env->isolate())
   };
-  MakeCallback(env, process_object, "emit", arraysize(args), args);
+  MakeCallback(env->isolate(),
+               process_object, "emit", arraysize(args), args,
+               0).ToLocalChecked();
```

Comment:
Good question. In theory, we should only have `0` as the id when the resource is unknown, which should only happen because of legacy code.

I'm not really sure what the appropriate resource is, I guess we could create an event-loop resource on startup.